### DISCUSSION TITLE
Create conda-forge package

### DISCUSSION THREAD

**Discussion by juanfcocontreras** [zauberzeug/nicegui#768] [Ideas / Feature Requests]
I think it would be good for the spread of the package, if it was available in the conda-forge repositories. Many companies and organizations use conda, and it is not good to mix pip and conda installations in the same environment, so many anaconda users choose not to install packages that are not available in conda.

The process is well documented and available from:

https://conda-forge.org/docs/maintainer/adding_pkgs.html

Do you think it would be possible?

**Comment by rodja** [comment]
Thanks for your suggestion @juanfcocontreras. Providing NiceGUI via conda-forge would be great so I've just created the pull request: https://github.com/conda-forge/staged-recipes/pull/22577.

**Comment by rodja** [comment]
It's not as easy as I had hoped. I get the conda build issues:

```
Encountered problems while solving:
  - nothing provides requested fastapi-socketio >=0.0.10,<0.0.11
  - nothing provides requested vbuild >=0.8.1,<0.9.0
  - nothing provides requested pywebview >=4.0.2,<4.1.0
```

Looks like the listed packages are not available on conda-forge yet. Can anybody help out?

**Reply by juanfcocontreras** [reply]
The problem is that when adding a package to conda-forge one should add both its package and those packages it depends on that are not already in conda-forge, and those 3 packages are not in coda-forge either. The solution would be to do exactly the same as you have done for nicegui for all three.

**Reply by rodja** [reply]
Thanks. Must these packages become supported "conda-forge" packages before NiceGUI? `pywebview` may be really hard. It depends on a lot of native stuff per platform. `fastapi-socketio` is trivial. I have not looked into `vbuild` but it may be possible.

**Reply by juanfcocontreras** [reply]
To be honest, I have never actually uploaded any packages to conda-forge, but I know there is a tool that supposedly makes it quite easy:

https://github.com/conda/grayskull

I have run it with pywebview and it generates a quite reasonable `meta.yaml` file, although it again has a couple more missing dependencies.

**Reply by rodja** [reply]
Yes. I've successfully used grayskull to create packages for `fastapi-socketio` and `vbuild`. `pywebview` is another matter. The dependencies are more tricky. Maybe it is better to ask the project to create the package themself?

**Reply by juanfcocontreras** [reply]
Yes, I think it is the most convenient thing to do. I will open an issue in your repository. Thank you very much.

**Comment by rodja** [comment]
I've removed the pywebview dependency for now which had lead to allow us into conda-forge feedstock: https://github.com/conda-forge/nicegui-feedstock with 1.2.7.

Unfortunaltely the auto-build 1.2.8 fails: https://github.com/conda-forge/nicegui-feedstock/pull/1 ([poetry not found](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=692305&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=187)).

**Reply by juanfcocontreras** [reply]
Removing that dependency seems to me the most appropriate, since many people will use nicegui without generating packages, and so at least they can have it available (although some note to indicate it may come in handy).

As for the problem, I think it could be solved by adding poetry in the list of dependencies. Can you try it?

**Reply by rodja** [reply]
That does not help: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=693104&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=288

Why is this soooo complicated?

**Reply by juanfcocontreras** [reply]
The truth is that I don't know, it seems that poetry and conda don't really get along. Here they recommend a solution, could you try it?

https://stackoverflow.com/a/72275016

> Make sure you:
>
> - use {{PYTHON}} instead of python
> - include poetry in both build and host requirements


Thank you very much for your patience!

**Reply by juanfcocontreras** [reply]
Here is a similar solution:

https://github.com/wasdee/thaifin/issues/3#issuecomment-1036080150

> by adding poetry to **host** dependencies:

```
requirements:
  host:
   - python
   - pip
   - poetry
```

**Comment by rodja** [comment]
Wonderful @juanfcocontreras. adding poetry to host and build dependencies worked. Now the build fails because it [can not find webview](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=695121&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=443). Because pywebview is [not yet buildable on conda-forge](https://github.com/zauberzeug/nicegui/discussions/768#discussioncomment-5657640) we should make sure to build/start without it.

**Reply by rodja** [reply]
I've made sure we can start without pywebview with 21a85a8. Hopefully we can successfully package on feedstock with the next release.

**Reply by juanfcocontreras** [reply]
Great @rodja , I'm really glad that nicegui is finally available in conda-forge. I have created an issue in pywebview:

https://github.com/r0x0r/pywebview/issues/1106

Hopefully they will also manage to add it to conda-forge.

Thanks a lot for everything!

**Comment by rodja** [comment]
We did it! NiceGUI 1.2.10 is now available on conda-forge: https://anaconda.org/conda-forge/nicegui
The feedstock is managed through this repo: https://github.com/conda-forge/nicegui-feedstock
